### ISSUE TITLE
test accuracy is evaluated with dropout

### ISSUE THREAD

**Issue by alexanderchr** [d2l-ai/d2l-en#1093]
In the torch code in section 4.6.4.2, an instance variable `is_training` is defined, but is kept constant during training, so the test accuracy is evaluated with dropout enabled which is probably not intended.

There is an instance variable `training` inherited from `nn.Module` (used by the `nn.Dropout` layer) that could be used instead. See https://pytorch.org/docs/stable/nn.html#torch.nn.Module.train.

**Comment by astonzhang** [comment]
@AnirudhDagar

**Comment by StevenJokes** [comment]
It doesn't matter. Still works in training mode

[Discuss]( (https://discuss.pytorch.org/t/what-does-nn-modules-train-true-train-false-do/4004))
>Set the training attribute of Module and subModules to True or False.
>As far as I know, when I’m using Dropout layers, it randomly sets some output values to zero in training mode, but not in evaluation mode.

check for [source code]
(https://github.com/pytorch/pytorch/blob/2cdbb140e62d80906c4a6b1944e7e36650ec54b1/torch/nn/modules/module.py#L978)

**Comment by AnirudhDagar** [comment]
@alexanderchr this has been addressed in #1110

**Comment by StevenJokes** [comment]
@AnirudhDagar `net.train()` doesn't need to be explicit, but it is better to be.But net.eval needs to be explicit, because it is not default.

**Comment by AnirudhDagar** [comment]
Yes, I know. But once you switch to net.eval(), you need to switch back to net.train() in the training loop.

**Comment by alexanderchr** [comment]
@AnirudhDagar Thanks! I think to fully resolve this issue, the class Net defined in section 4.6.4.2 should be changed to use `self.training` instead of `self.is_training`.

**Comment by AnirudhDagar** [comment]
@alexanderchr that is the custom model we have built from scratch and hence it doesn't relate to how PyTorch's `net.train` sets `self.training=True`. Hence it is perfectly okay to use `self.is_training`.

On the other hand, for the concise implementation, we do need to use this, hence, I've fixed the issue already.

**Comment by alexanderchr** [comment]
@AnirudhDagar: of course you can call it anything, but as it stands the value of ‘net.is_training’ is never changed. So either ‘train_ch3’ need to set the value of ‘net.is_training’ (in addition to calling ‘net.train()/eval()’) or the custom model should be modified to use ‘self.training’ instead.

**Comment by AnirudhDagar** [comment]
Yes @alexanderchr  Thanks for pointing this out.

**Comment by AnirudhDagar** [comment]
Yes @alexanderchr Thanks a tonne, for pointing this out.
